extensions/statuspage-monitor/src/viewStatus.tsx

```diff
@@ -0,0 +1,118 @@
+import { ActionPanel, List, Action, Icon, Color } from "@raycast/api";
+
+import { useState, useEffect } from "react";
+import AddServiceList from "./addService";
+import { getStatus } from "./lib/api";
+import { getPageIds, removePageId } from "./lib/store";
+import { PageStatus } from "./lib/types";
+import { capitalizeFirstLetter, iconForIndicator } from "./lib/util";
+import ServiceStatusList from "./service";
+
+export default function ViewStatusList() {
+  const [isLoading, setLoading] = useState(true);
+  const [services, setServices] = useState<PageStatus[]>([]);
+  const [missingServices, setMissingServices] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (!services.length) fetchPageData();
+  }, []);
+
+  async function fetchPageData() {
+    setLoading(true);
+
+    const pageIds = await getPageIds();
+
+    const missing: string[] = [];
+    const fetchStatusPages = await Promise.all(
+      pageIds.map(
+        (id) =>
+          new Promise<PageStatus | null>((res) => {
+            getStatus(id)
+              .then(res)
+              .catch(() => {
+                missing.push(id);
+                res(null);
+              });
+          })
+      )
+    );
+
+    const pages = fetchStatusPages.filter((p): p is PageStatus => !!p);
+
+    setServices(pages);
+    setMissingServices(missing);
+    setLoading(false);
+  }
+
+  function AddServicePushActionPanel() {
+    return (
+      <ActionPanel>
+        <Action.Push title="Add Service" icon={Icon.Plus} target={<AddServiceList refreshStatus={fetchPageData} />} />
+      </ActionPanel>
+    );
+  }
+
+  function ServiceActionPanel({ id }: { id: string }) {
+    return (
+      <ActionPanel>
+        <Action.Push title="View Status" icon={Icon.Clock} target={<ServiceStatusList id={id} />} />
+        <Action.OpenInBrowser title="Visit Statuspage" url={`https://${id}.statuspage.io`} />
```

Comment:
Why not link to the official status page of the service? At least when the user opens it in the browser. For example for GitHub instead of going to: https://kctbh9vrtdwd.statuspage.io/ it will go to https://www.githubstatus.com/